src/GraphQL/Execution/SubscriptionExecutionStrategy.cs

```diff
@@ -0,0 +1,310 @@
+                            {
+                                node,
+                            }
+                        };
+                    }
+                }
+            }
+            catch (OperationCanceledException) when (context.CancellationToken.IsCancellationRequested)
+            {
+                // generate a static error message here;
+                // do not "throw;" back to the caller (the event source)
+                result.AddError(GenerateError(context, node, "The operation was canceled."));
+            }
+            catch (ExecutionError executionError)
+            {
+                executionError.Path = node.ResponsePath;
+                executionError.AddLocation(node.Field, context.Document);
+                result.AddError(executionError);
+            }
+            catch (Exception exception)
+            {
+                result.AddError(await HandleExceptionInternalAsync(context, node, exception,
+                    $"Could not process event data for field '{node.Field.Name}'.").ConfigureAwait(false));
+            }
+
+            result.AddErrors(context.Errors);
+
+            return result;
+        }
+
+        /// <summary>
+        /// Encapsulates an error within an <see cref="ExecutionResult"/> for errors generated
+        /// by the event stream via <see cref="IObserver{T}.OnError(Exception)"/>.
+        /// </summary>
+        protected virtual Task<ExecutionError> ProcessErrorAsync(ExecutionContext context, ExecutionNode node, Exception exception)
+            => HandleExceptionInternalAsync(context, node, exception, $"Event stream error for field '{node.Field.Name}'.");
+
+        /// <summary>
+        /// Generates an <see cref="ExecutionError"/> for the specified <see cref="Exception"/>
+        /// and sets the <see cref="ExecutionError.Path"/> and <see cref="ExecutionError.Locations"/> properties.
+        /// </summary>
+        private async Task<ExecutionError> HandleExceptionInternalAsync(ExecutionContext context, ExecutionNode node, Exception exception, string defaultMessage)
+        {
+            var executionError = await HandleExceptionAsync(context, node, exception, defaultMessage).ConfigureAwait(false);
+            executionError.Path = node.ResponsePath;
+            executionError.AddLocation(node.Field, context.Document);
+            return executionError;
+        }
+
+        /// <summary>
+        /// Generates an <see cref="ExecutionError"/> for the specified <see cref="Exception"/>.
+        /// </summary>
+        protected virtual async Task<ExecutionError> HandleExceptionAsync(ExecutionContext context, ExecutionNode node, Exception exception, string defaultMessage)
+        {
+            UnhandledExceptionContext? exceptionContext = null;
+            if (context.UnhandledExceptionDelegate != null)
+            {
+                exceptionContext = new UnhandledExceptionContext(context, new ReadonlyResolveFieldContext(node, context), exception);
```

Comment:
```suggestion
                exceptionContext = new UnhandledExceptionContext(context, new ReadonlyResolveFieldContext(node, context), exception) { ErrorMessage = defaultMessage };
```

Reply:
This does not match any other exception handling in the project.  `ErrorMessage` is initially `null`.  `Exception` is the original exception.  If `Exception` is set to an instance of `ExecutionError`, then it is returned.  If not, it is wrapped in an `UnhandledError` and the message is set to `ErrorMessage` if set, or the default message if not set.

Reply:
So the code here lines up to `ExecutionStrategy.ProcessNodeUnhandledExceptionAsync` and the code within the `DocumentExecuter` and I do not believe it should be changed.

Reply:
OK, I see.